orangecontrib/text/widgets/owtopicmodeling.py

```diff
@@ -0,0 +1,250 @@
+            method_layout.addWidget(widget)
+        method_layout.addStretch()
+        self.toggle_widgets()
+        self.group.buttonClicked.connect(self.change_method)
+
+        # Commit button
+        gui.auto_commit(self.buttonsArea, self, 'autocommit', 'Commit', box=False)
+
+        # Topics description
+        self.cols = ['Topic', 'Topic keywords']
+        self.topic_desc = QtGui.QTreeWidget()
+        self.topic_desc.setColumnCount(len(self.cols))
+        self.topic_desc.setHeaderLabels(self.cols)
+        # self.topic_desc.setSelectionMode(QtGui.QTreeView.ExtendedSelection)
+        self.topic_desc.itemSelectionChanged.connect(self.selected_topic_changed)
+        for i in range(len(self.cols)):
+            self.topic_desc.resizeColumnToContents(i)
+        self.mainArea.layout().addWidget(self.topic_desc)
+
+    def set_data(self, data=None):
+        self.corpus = data.copy()
+        self.apply()
+
+    def commit(self):
+        if self.corpus is not None:
+            self.apply()
+
+    @property
+    def model(self):
+        return self.widgets[self.method_index].model
+
+    def change_method(self):
+        if self.method_index != self.group.checkedId():
+            self.method_index = self.group.checkedId()
+            self.toggle_widgets()
+            self.commit()
+
+    def toggle_widgets(self):
+        for i, widget in enumerate(self.widgets):
+            widget.setVisible(i == self.method_index)
+
+    def update_topics(self):
+        self.topic_desc.clear()
+        for i in range(self.model.num_topics):
+            words = self.model.get_top_words_by_id(i)
+            it = LDATreeWidgetItem(i, words, self.topic_desc)
+            self.topic_desc.addTopLevelItem(it)
+        for i in range(2):
+            self.topic_desc.resizeColumnToContents(i)
+
+    def selected_topic_changed(self):
+        selected = self.topic_desc.selectedItems()
+        if selected:
+            self.send_topic_by_id(selected[0].topic_id)
+
+    def send_topic_by_id(self, topic_id):
+        self.topic_desc.setCurrentItem(self.topic_desc.topLevelItem(topic_id))
+        self.send(Output.TOPICS, self.model.get_topics_table_by_id(topic_id))
+
+    def enabled(self, bool):
+        pass
+
+    def progress(self, p):
+        self.progressBarSet(p)
+
+    def apply(self):
+        self.topic_desc.clear()
+        if self.corpus:
+            self.enabled(False)
+
+            with self.progressBar(len(self.corpus)) as bar:
+                self.model.fit(self.corpus)
+                table = self.model.transform(self.corpus)
+                self.update_topics()
+
+            self.send(Output.DATA, table)
+            self.send_topic_by_id(0)
+
+            self.enabled(True)
+        else:
+            self.send(Output.DATA, None)
+            self.send(Output.TOPICS, None)
+
+    def send_report(self):
+        self.report_items(self.model.name, ())
+
+
+class LDATreeWidgetItem(QtGui.QTreeWidgetItem):
```

Comment:
Since we support many topic models now we could rename this to something not starting with LDA 😉